src/VisualStudio/Core/Def/Implementation/TodoComments/VisualStudioTodoCommentsService.cs

```diff
@@ -114,19 +115,18 @@ private async Task StartWorkerAsync()
 
             // Pass ourselves in as the callback target for the OOP service.  As it discovers
             // todo comments it will call back into us to notify VS about it.
-            _connection = await client.CreateConnectionAsync(
-                WellKnownServiceHubService.RemoteTodoCommentsService,
-                callbackTarget: this, cancellationToken).ConfigureAwait(false);
+            var connection = await client.CreateConnectionAsync<IRemoteTodoCommentsService>(callbackTarget: this, cancellationToken).ConfigureAwait(false);
 
             // Now that we've started, let the VS todo list know to start listening to us
             _eventListenerTracker.EnsureEventListener(_workspace, this);
 
             // Now kick off scanning in the OOP process.
-            await _connection.RunRemoteAsync(
-                nameof(IRemoteTodoCommentsService.ComputeTodoCommentsAsync),
-                solution: null,
-                arguments: Array.Empty<object>(),
+            // If the call fails an error has already been reported and there is nothing more to do.
+            _ = await connection.TryInvokeAsync(
+                (service, cancellationToken) => service.ComputeTodoCommentsAsync(cancellationToken),
                 cancellationToken).ConfigureAwait(false);
+
+            _lazyConnection = connection;
```

Comment:
1. if you're intentionally setting this at the end, state why that's important.  otherwise, set on the line you create the connection.
2. rename `_lazyConnection` to `_connection`.

Reply:
The connection is lazy though as it is initialized on demand.

Reply:
Will do [1]